numpy/ma/core.py

```diff
@@ -3316,6 +3317,11 @@ def _scalar_heuristic(arr, elem):
                                 f"all to {dout._fill_value[0]!s}.",
                                 stacklevel=2)
                         dout._fill_value = dout._fill_value.flat[0]
+                    elif not hasattr(dout._fill_value, 'ndim'):
+                        # only initialize to empty array if it dout does not 
+                        # have ._fill_value before. If it has, do not change 
+                        # the original behavior
+                        dout._fill_value = np.array([], dtype)
```

Comment:
This doesn't strike me as correct, `_fill_value` should always be a 0d array, this is a 1d array.

Reply:
Oops sorry I'll look into that!

Reply:
I suspect you want
```suggestion
                        dout._fill_value = np.empty((), dtype)
                        dout._fill_value[()] = self._fill_value[indx]
```

Reply:
Yes thank you! I'll try to add some tests for this in the next commit!